pep-9999.rst

```diff
@@ -0,0 +1,173 @@
+PEP: 9999
+Title: Core support for generic types
+Author: Ivan Levkivskyi <levkivskyi@gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 03-Sep-2017
+Post-History: 09-Sep-2017
+
+
+Abstract
+========
+
+Initially PEP 484 was designed in such way that it would not introduce
+*any* changes to the core CPython interpreter. Now that type hints and
+the ``typing`` module are extensively used by the community, this restriction
+can be removed. It is proposed to add two special methods
```

Comment:
I wonder if PEP 526 (variable annotations) and PEP 557 (dataclasses) should also be mentioned as an indication that type hints are becoming mainstream.